osf/models/node.py

```diff
@@ -1240,6 +1240,8 @@ def add_contributor_registered_or_not(self, auth, user_id=None,
             contributor = OSFUser.load(user_id)
             if not contributor:
                 raise ValueError('User with id {} was not found.'.format(user_id))
+            if not contributor.is_registered:
+                raise ValueError('Cannot add unconfirmed user {} to node {}'.format(user_id, self._id))
```

Comment:
I think this is not entirely correct. You can add an unconfirmed user, but you have to do it another way. Can you indicate the proper method in this error response?